lib/fluent/root_agent.rb

```diff
@@ -71,19 +71,41 @@ def configure(conf)
           raise ConfigError, "Missing worker id on <worker> directive"
         end
 
-        target_worker_id = target_worker_id_str.to_i
-        if target_worker_id < 0 || target_worker_id > (Fluent::Engine.system_config.workers - 1)
-          raise ConfigError, "worker id #{target_worker_id} specified by <worker> directive is not allowed. Available worker id is between 0 and #{(Fluent::Engine.system_config.workers - 1)}"
-        end
+        target_worker_ids = target_worker_id_str.split("-")
+        if target_worker_ids.size == 2
```

Comment:
Could you add duplication check to avoid following case?

```
<worker 0-3>
</worker>

<worker 3-7> # should be 4-7
</worker>
```

Reply:
I've added a collisions checker in https://github.com/fluent/fluentd/pull/2292/commits/d900a736d11320a954a2b0482f71b80ca023b64d.

Reply:
And I added more concreate error message commit: https://github.com/fluent/fluentd/pull/2292/commits/03da2b3b51fcc6ccd9a435a51e73703dd2d1f208.